samples/CosmosDBIntegration/CosmosDbIntegrationTutorial.md

```diff
@@ -0,0 +1,117 @@
+    - [Get started with Azure Functions triggers in Azure Cache for Redis](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-tutorial-functions-getting-started)
+        - Finish steps up to (and including), "Configure the cache."
+
+- Sample code
+    - ReadThrough
+        - [PubSub.Sample](ReadThroughSamples/PubSubSample.cs)
+        - [List.Sample](ReadThroughSamples/ListSample.cs)
+    - WriteAround
+        - [PubSub.Sample](WriteAroundSamples/PubSubSample.cs)
+        - [List.Sample](WriteAroundSamples/ListSample.cs)
+        - [Stream.Sample](WriteAroundSamples/StreamSample.cs)
+    - WriteBehind
+        - [PubSub.Sample](WriteBehindSamples/PubSubSample.cs)
+        - [List.Sample](WriteBehindSamples/ListSample.cs)
+        - [Stream.Sample](WriteBehindSamples/StreamSample.cs)
+    - WriteThrough
+        - [PubSub.Sample](WriteThroughSamples/PubSubSample.cs)
+        - [Stream.Sample](WriteThroughSamples/StreamSample.cs)
+
+## Installing Packages and adjusting variables
+
+### Install packages and adding code
+1. In the terminal, add the following packages
+    1. dotnet add package Microsoft.Azure.Cosmos
+    2. dotnet add package Microsoft.Azure.WebJobs.Extensions.CosmosDB
+    3. dotnet add package Microsoft.Azure.WebJobs.Extensions.Redis --prerelease
+    4. dotnet add package Microsoft.NET.Sdk.Functions
+2. Choose a sample from the Prerequisites to use with your Azure Function according to each trigger and caching pattern.
+3. Copy and paste the sample code into your azure function, using a cs file.
+
+### Adding connection string and variables
+1. Direct to your local.settings.json file in Visual Studio.
+2. Copy and paste the following code as a replacement for your local.settings.json.
+3. Replace each variable with the appropriate value.
+ ```json
+{
+  "IsEncrypted": false,
+  "Values": {
+    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
+    "FUNCTIONS_WORKER_RUNTIME": "dotnet",
+    "CosmosDbDatabaseId": "<database-id>",
+    "ListCosmosDbContainerId": "<container-id-for-list>",
+    "PubSubCosmosDbContainerId": "<container-id-for-pubsub>",
+    "MessagesCosmosDbContainerId": "<container-id-for-pubsub-messages>",
+    "PubSubChannel": "<pubsub-channel-name>",
+    "StreamCosmosDbContainerId": "<container-id-for-stream>",
+    "StreamCosmosDbContainerIdSingleDocument": "<container-id-for-stream-singular-document>",
+    "StreamTest": "<stream-name>",
+    "StreamTestSingleDocument": "<stream-name-singular-document>",
+    "RedisConnectionString": "<cache-name>.redis.cache.windows.net:6380,password=<access-key>,ssl=True,abortConnect=False,tiebreaker=",
+    "CosmosDbConnectionString": "AccountEndpoint=https://<cosmosdb-account>.documents.azure.com:443/;AccountKey=<access-key>;"
+  }
+}
+ ```
+ NOTE: each trigger should be using a separate container, which is dependent on the variable names.
+
+ #### Description of each variable
```

Comment:
This is a good thing to have because the code above assumes a lot. Could be a table too. I hate tables but most people love them. I hate them because when you look at them in Markdown you can get vertigo (JK - but they can really suck to look at).